pkg/certificate/fake_manager.go

```diff
@@ -0,0 +1,39 @@
+package certificate
+
+import (
+	"fmt"
+	time "time"
+)
+
+var (
+	caCert = &Certificate{
+		CommonName: "Test CA",
+		Expiration: time.Now().Add(time.Hour * 24),
+	}
+	validity = time.Hour
+)
+
+type fakeIssuer struct{}
+
+//IssueCertificate is a testing helper to satisfy the certificate client interface
+func (i *fakeIssuer) IssueCertificate(cn CommonName, validityPeriod time.Duration) (*Certificate, error) {
+	return &Certificate{
+		CommonName: cn,
+		Expiration: time.Now().Add(validityPeriod),
+	}, nil
+}
+
+//FakeCertManager is a testing helper that returns a *certificate.Manager
+func FakeCertManager() (*Manager, error) {
+	cm, err := NewManager(
+		caCert,
+		&fakeIssuer{},
+		validity,
+		nil,
+		make(chan struct{}),
```

Comment:
Creating the channel here means nothing calling this from outside the package is able to access it to close it and stop the goroutine rotating certs. I'm not sure exactly what happens when you create this in a test and don't stop cert rotation, but I imagine it probably doesn't matter unless we're creating 1000s of these in a single test which is definitely not the case.